core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -89,17 +89,26 @@ protected void killComputer(Computer c) {
         c.kill();
     }
 
-    /* =================================================================================================================
-    * Package-protected, but accessed API
-    * ============================================================================================================== */
-
-    /*package*/ final CopyOnWriteArraySet<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<>();
+    private Set<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<>();
 
-    @Restricted(NoExternalUse.class)
-    public CopyOnWriteArraySet<String> getDisabledAdministrativeMonitors(){
+    /**
+     * Get the disabled administrative monitors
+     *
+     * @since TODO
+     */
+    public Set<String> getDisabledAdministrativeMonitors(){
     	return disabledAdministrativeMonitors;
     }
-    
+
+    /**
+     * Set the disabled administrative monitors
+     *
+     * @since TODO
+     */
+    public void setDisabledAdministrativeMonitors(Set<String> disabledAdministrativeMonitors) {
```

Comment:
```suggestion
    public synchronized void setDisabledAdministrativeMonitors(Set<String> disabledAdministrativeMonitors) {
```

Reply:
why the instance-level lock suggestion, vs. what's been implemented here. seems like overkill

Reply:
This is outdated